lib/services/role.go

```diff
@@ -147,7 +147,7 @@ func NewAdminRole() Role {
 func NewImplicitRole() Role {
 	return &RoleV3{
 		Kind:    KindRole,
-		Version: V3,
+		Version: types.V4,
```

Comment:
This being V4 role now - does nothing break with implicit role not having any allowed labels set?

Reply:
well so far nothing broke but thanks for pointing this out, the plan for this PR was actually to leave the builtin roles on V3 so I've reverted the change here and in RoleForUser